File: app/src/main/java/com/helin/noteapplication2/NotesFragment.java
package com.helin.noteapplication2;

import android.app.Activity;
import android.app.Dialog;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.database.Cursor;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.Uri;
import android.os.Bundle;
import android.provider.MediaStore;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.helin.noteapplication2.databinding.FragmentNotesBinding;

import java.util.ArrayList;

import static android.app.Activity.RESULT_OK;
import static androidx.core.app.ActivityCompat.startActivityForResult;
import static com.helin.noteapplication2.Adapter.myDialog;

public class NotesFragment extends Fragment {
    RecyclerView recyclerView;
    private Adapter adapter;
    private Button addNote;
    DBHelper db;
    Dialog dialogadd;
    Dialog dialogedit;
    RadioButton radioButton;
    Person person;
    private ArrayList<Notes> notes = new ArrayList<>();
    private TextView info;
    public static final int PICK_IMAGE = 5;
    public static final int READ_EXTERNAL_STORAGE = 101;//private  binding;
    ImageView uploadImg;
    public static String selectedImagePath;
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        FragmentNotesBinding binding = FragmentNotesBinding.inflate(inflater, container, false);
        db = new DBHelper(getContext());
        db.getWritableDatabase();
        info = binding.textViewInfo;
        addNote = binding.buttonAddNote;
        recyclerView = binding.RecyclerViewNotes;

        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        myDialog = new Dialog(getContext(), android.R.style.Theme_DeviceDefault_Light_NoActionBar_Fullscreen);
        myDialog.setContentView(R.layout.dialog_box_edit);
        uploadImg=myDialog.findViewById(R.id.imageViewUpload);


        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(requireContext());
        linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
        recyclerView.setLayoutManager(linearLayoutManager);
        person = NotesFragmentArgs.fromBundle(getArguments()).getPerson();
        notes = db.getAllNotes(person.getId());


        adapter = new Adapter(NotesFragment.this, getContext(), notes, person);
        recyclerView.setAdapter(adapter);
        Log.e("personget", person.toString());
        info.setText("Welcome " + person.getName() + " " + person.getSurname());
        addNote.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dialogAdd();
            }
        });


        return binding.getRoot();
    }


    public void dialogAdd() {
        RadioGroup rg;
        Button okayBtn;

        dialogadd = new Dialog(getContext(), android.R.style.Theme_DeviceDefault_Light_NoActionBar_Fullscreen);

        dialogadd.setContentView(R.layout.dialog_box_add);
        rg = dialogadd.findViewById(R.id.RadioGroupNoteAdd);
        okayBtn = dialogadd.findViewById(R.id.buttonOkay);


        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext());
        linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
        dialogadd.show();
        okayBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                NotesDao itemAdd = new NotesDao(getContext());
                EditText note = dialogadd.findViewById(R.id.editTextEnterNote);
                int type = rg.getCheckedRadioButtonId();
                radioButton = dialogadd.findViewById(type);
                if (radioButton.getText().toString().contains("ToDo")) {
                    itemAdd.addNote(db, NoteType.TODO, note.getText().toString(), person.getId());
                } else if (radioButton.getText().toString().contains("Done")) {
                    itemAdd.addNote(db, NoteType.DONE, note.getText().toString(), person.getId());
                } else if (radioButton.getText().toString().contains("In Progress")) {
                    itemAdd.addNote(db, NoteType.INPROGRESS, note.getText().toString(), person.getId());
                }

                dialogadd.dismiss();
                notes.clear();
                notes.addAll(db.getAllNotes(person.getId()));
                adapter.notifyDataSetChanged();

            }
        });

    }

    @Override
    public void onRequestPermissionsResult(int requestCode,
                                           @NonNull String[] permissions,
                                           @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode,
                permissions,
                grantResults);

        if (requestCode == READ_EXTERNAL_STORAGE) {
            if (grantResults.length > 0
                    && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                Toast.makeText(getContext(), "Storage Permission Granted", Toast.LENGTH_SHORT).show();

            } else {
                Toast.makeText(getContext(), "Storage Permission Denied", Toast.LENGTH_SHORT).show();
            }
        }
    }
    public void onActivityResult(int requestCode, int resultCode, @Nullable @org.jetbrains.annotations.Nullable Intent data) {
        if (requestCode == PICK_IMAGE && resultCode == RESULT_OK) {
            Uri selectedImageUri = data.getData();
            selectedImagePath = getRealPathFromURIForGallery(selectedImageUri);
            //Log.d("onActivityResult: "+selectedImagePath);
            Bitmap myBitmap = BitmapFactory.decodeFile(selectedImagePath);
            uploadImg.setImageBitmap(myBitmap);

           //// notes.set(selectedImagePath);
            //binding.textViewImagePath.setText(selectedImagePath);
        }
    }
    public  void selectImage() {

        Intent intent = new Intent(Intent.ACTION_PICK,
                android.provider.MediaStore.Images.Media.EXTERNAL_CONTENT_URI);

        startActivityForResult(Intent.createChooser(intent, "select image"),
                PICK_IMAGE);
    }
    public String getRealPathFromURIForGallery(Uri uri) {
        if (uri == null) {
            return null;
        }
        String[] projection = {MediaStore.Images.Media.DATA};
        Cursor cursor = requireContext().getContentResolver().query(uri, projection, null,
                null, null);
        if (cursor != null) {
            int column_index =
                    cursor.getColumnIndexOrThrow(MediaStore.Images.Media.DATA);
            cursor.moveToFirst();
            return cursor.getString(column_index);
        }
        assert false;
        cursor.close();
        return uri.getPath();
    }




}

File: app/src/main/java/com/helin/noteapplicationkotlin/NotesFragment.kt
package com.helin.noteapplicationkotlin

import android.app.Activity
import android.app.Dialog
import android.content.ContentValues
import android.content.Intent
import android.content.pm.PackageManager
import android.graphics.BitmapFactory
import android.net.Uri
import android.os.Bundle
import android.provider.MediaStore
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.helin.noteapplicationkotlin.databinding.FragmentNotesBinding
import java.io.FileNotFoundException
import java.util.*
import kotlin.collections.ArrayList

class NotesFragment : Fragment() {
    var recyclerView: RecyclerView? = null
    private var adapter: Adapter? = null
    private var addNote: Button? = null
    var db: DBHelper? = null
    var dialogadd: Dialog? = null
    lateinit var radioButton: RadioButton
    var person: Person? = null
    private var notes: ArrayList<Notes> = ArrayList()
    private var info: TextView? = null
    var uploadImg: ImageView? = null
    var noteList=ArrayList<Notes>()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val binding = FragmentNotesBinding.inflate(inflater, container, false)
        val model: NotesViewModel by viewModels()

        info = binding.textViewInfo
        addNote = binding.buttonAddNote
        recyclerView = binding.RecyclerViewNotes

        init()

        Log.e("personget", person.toString())

        info!!.text = "Welcome " + person!!.name + " " + person!!.surname

        addNote!!.setOnClickListener {

            model.addNote(person!!,requireContext())
        }
        model.getAllNotes(requireContext(), person!!)
        model.notes.observe(viewLifecycleOwner, androidx.lifecycle.Observer {
            noteList.clear()
            noteList.addAll(it)
            Log.e("notes", "onCreateView: "+it.size)
            binding.RecyclerViewNotes.adapter!!.notifyDataSetChanged()

        })
        adapter= Adapter(this,requireContext(),noteList, person!!)
        binding.RecyclerViewNotes.adapter=adapter
        return binding.root
    }


    override fun onRequestPermissionsResult(requestCode: Int,
                                            permissions: Array<String>,
                                            grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode,
                permissions,
                grantResults)
        if (requestCode == READ_EXTERNAL_STORAGE) {
            if (grantResults.size > 0
                    && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                Toast.makeText(context, "Storage Permission Granted", Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(context, "Storage Permission Denied", Toast.LENGTH_SHORT).show()
            }
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if (requestCode == PICK_IMAGE && resultCode == Activity.RESULT_OK) {
            val selectedImageUri = data!!.data
            selectedImagePath = getRealPathFromURIForGallery(selectedImageUri)
            try {
                val bitmap = BitmapFactory.decodeStream(requireActivity().contentResolver.openInputStream(selectedImageUri!!))
                uploadImg!!.setImageBitmap(bitmap)
            } catch (e: FileNotFoundException) {
                Log.e(ContentValues.TAG, "onActivityResult: " + e.localizedMessage)
                e.printStackTrace()
            }
        }
    }

    fun selectImage() {
        val intent = Intent(Intent.ACTION_PICK,
                MediaStore.Images.Media.EXTERNAL_CONTENT_URI)
        startActivityForResult(Intent.createChooser(intent, "select image"),
                PICK_IMAGE)
    }

    fun getRealPathFromURIForGallery(uri: Uri?): String? {
        if (uri == null) {
            return null
        }
        val projection = arrayOf(MediaStore.Images.Media.DATA)
        val cursor = requireContext().contentResolver.query(uri, projection, null,
                null, null)
        if (cursor != null) {
            val column_index = cursor.getColumnIndexOrThrow(MediaStore.Images.Media.DATA)
            cursor.moveToFirst()
            return cursor.getString(column_index)
        }
        assert(false)
        if (cursor != null) {
            cursor.close()
        }
        return uri.path
    }

    fun init() {

        recyclerView?.layoutManager = LinearLayoutManager(context)
        Adapter.Companion.myDialog = Dialog(requireContext(), android.R.style.Theme_DeviceDefault_Light_NoActionBar_Fullscreen)
        Adapter.Companion.myDialog!!.setContentView(R.layout.dialog_box_edit)
        uploadImg = Adapter.Companion.myDialog!!.findViewById<ImageView>(R.id.imageViewUpload)
        val linearLayoutManager = LinearLayoutManager(requireContext())
        linearLayoutManager.orientation = LinearLayoutManager.VERTICAL
        recyclerView?.layoutManager = linearLayoutManager
        person = NotesFragmentArgs.fromBundle(requireArguments()!!).person


    }

    companion object {
        const val PICK_IMAGE = 5
        const val READ_EXTERNAL_STORAGE = 101
        var selectedImagePath: String? = null
    }
}